src/main/java/htmlpublisher/HtmlPublisherTarget.java

```diff
@@ -221,18 +225,28 @@ private Run getArchiveBuild(AbstractProject abstractProject) {
         protected String getTitle() {
             return this.project.getDisplayName() + " html2";
         }
+        
+        /**
+         * Gets {@link HtmlPublisherTarget}, for which the action has been created.
+         * @return HTML Report description
+         * @since TODO
+         */
+        public @Nonnull HtmlPublisherTarget getHTMLTarget() {
+            return HtmlPublisherTarget.this;
+        }
     }
 
-    public class HTMLBuildAction extends BaseHTMLAction {
-        private final AbstractBuild<?, ?> build;
+    public class HTMLBuildAction extends BaseHTMLAction implements RunAction2 {
+        private transient Run<?, ?> build;
 
-        public HTMLBuildAction(AbstractBuild<?, ?> build, HtmlPublisherTarget actualHtmlPublisherTarget) {
+        public HTMLBuildAction(Run<?, ?> build, HtmlPublisherTarget actualHtmlPublisherTarget) {
             super(actualHtmlPublisherTarget);
             this.build = build;
         }
         
-        public final AbstractBuild<?,?> getOwner() {
-        	return build;
+        @WithBridgeMethods(value = AbstractBuild.class, castRequired = true)
```

Comment:
This probably works, though I have found it to be more trouble than it is worth, compared to simply introducing a new method and deprecating the old.